engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java

```diff
@@ -667,6 +659,15 @@ protected void advanceExpunge(VMInstanceVO vm) throws ResourceUnavailableExcepti
 
     }
 
+    private void addAllExpungeCommandsFromList(List<Command> cmdList, Commands cmds, VMInstanceVO vm) {
+        if (CollectionUtils.isNotEmpty(cmdList)) {
```

Comment:
We could invert the logic here, to reduce code indentation, and maybe add a log in trace.

Reply:
Thanks @GutoVeronezi, done